metadata-ingestion/src/datahub/ingestion/graph/client.py

```diff
@@ -1199,6 +1199,126 @@ def create_tag(self, tag_name: str) -> str:
         # return urn
         return res["createTag"]
 
+    def _assertion_result_shared(self) -> str:
+        fragment: str = """
+             fragment assertionResult on AssertionResult {
+                 type
+                 rowCount
+                 missingCount
+                 unexpectedCount
+                 actualAggValue
+                 externalUrl
+                 nativeResults {
+                     value
+                 }
+                 error {
+                     type
+                     properties {
+                         value
+                     }
+                 }
+             }
+        """
+        return fragment
+
+
+    def _run_assertion_result_shared(self) -> str:
+        fragment: str = """
+            fragment runAssertionResult on RunAssertionResult {
+                assertion {
+                    urn
+                }
+                result {
+                    ... assertionResult
+                }
+            }
+        """
+        return fragment
+
+
+    def run_assertion(self, urn: str, save_result: bool = False):
+        graph_query: str = """
+            %s
+            mutation runAssertion($assertionUrn: String!, $saveResult: Boolean) {
+                runAssertion(urn: $assertionUrn, saveResult: $saveResult) {
+                    ... assertionResult
+                }
+            }
+        """ % (self._assertion_result_shared())
+
+        variables = {
+            "assertionUrn": urn,
+            "saveResult": save_result,
+        }
+
+        res = self.execute_graphql(
+            query=graph_query,
+            variables=variables,
+        )
+
+        # return urn
+        return res["runAssertion"]
+
+
+    def run_assertions(self, urns: List[str], save_result: bool = False):
+        graph_query: str = """
+            %s
+            %s
+            mutation runAssertions($assertionUrns: [String!]!, $saveResult: Boolean) {
+                runAssertions(urns: $assertionUrns, saveResults: $saveResult) {
+                    passingCount
+                    failingCount
+                    errorCount
+                    results {
+                        ... runAssertionResult
+                    }
+                }
+            }
+        """ % (self._assertion_result_shared(), self._run_assertion_result_shared())
+
+        variables = {
+            "assertionUrns": urns,
+            "saveResult": save_result,
+        }
+
+        res = self.execute_graphql(
+            query=graph_query,
+            variables=variables,
+        )
+
+        # return urn
+        return res["runAssertions"]
+
+
+    def run_assertions_for_asset(self, urn: str):
```

Comment:
Please add optional parameter of array tagUrns as well! We support running only the assertions with a specific tag